src/components/shared/TransactionLink/TransactionLink.tsx

```diff
@@ -1,49 +1,28 @@
-import React from 'react';
-import { type MessageDescriptor } from 'react-intl';
+import { createElement, type FC } from 'react';
 
 import ExternalLink from '~shared/ExternalLink/index.ts';
-import { type SimpleMessageValues } from '~types/index.ts';
 import { getBlockExplorerLink } from '~utils/external/index.ts';
 
-interface Props {
-  /** Allows for link style customization (Eg: we need to disguise the link as a button)
-   * Don't abuse it!
-   */
-  className?: string;
-
-  /** Transaction hash */
-  hash: string;
-
-  /** A string or a `messageDescriptor` that make up the link's text. Defaults to `hash`. */
-  text?: MessageDescriptor | string;
-
-  /** Values for text (react-intl interpolation) */
-  textValues?: SimpleMessageValues;
-
-  /**  Show a title over the element, on hover. This is browser native. */
-  title?: string;
-}
+import { type TransactionLinkProps } from './types.ts';
 
 const displayName = 'TransactionLink';
 
-const TransactionLink = ({
-  className,
+const TransactionLink: FC<TransactionLinkProps> = ({
   hash,
-  text,
-  textValues,
-  title,
-}: Props) => (
-  <ExternalLink
-    className={className}
-    href={getBlockExplorerLink({
-      linkType: 'tx',
-      addressOrHash: hash,
-    })}
-    text={text || hash}
-    textValues={textValues}
-    title={title}
-  />
-);
+  children,
+  ...rest
+}) =>
+  createElement(
+    ExternalLink,
+    {
+      href: getBlockExplorerLink({
+        linkType: 'tx',
+        addressOrHash: hash,
+      }),
+      ...rest,
+    },
+    children,
+  );
```

Comment:
I know it's practically the same but was just curious why `createElement` was preferred here instead of JSX?

Reply:
https://reactpatterns.js.org/docs/spreading-props-on-dom-elements-is-an-anti-pattern/ 

As spreading props to jsx is considered as antipattern for components like HOCs we can use createElement

Reply:
thank you, @iamsamgibbs! I just updated my PR.